### ISSUE TITLE
why is the ent_coef  so big

### ISSUE THREAD

**Issue by war3gu** [AI4Finance-Foundation/FinRL#476]
I use the super params as below
ticket_list = ['600000.SH', '600009.SH', '600016.SH']

train_start_date = '2015-01-01'
train_stop_date = '2019-08-01'
val_start_date = '2019-08-01'
val_stop_date = '2021-01-03'

    total_timesteps = 500000 

    env_kwargs_train = {
        "stock_dim": stock_dimension,
        "hmax": 1000,
        "initial_amount": 1000000,
        "buy_cost_pct": 6.87e-5,
        "sell_cost_pct": 1.0687e-3,
        "reward_scaling": 1e0,
        "state_space": state_space,
        "action_space": stock_dimension,
        "tech_indicator_list": config.TECHNICAL_INDICATORS_LIST,
        "print_verbosity": 100,  
        "initial_buy": True,
        "hundred_each_trade": True,
        "out_of_cash_penalty": 0.001,
        "cash_limit": 0.2,
        "model_name":"stock_a",
        "mode":"train",  
        "random_start":True
    }

    SAC_PARAMS = {
        "batch_size": 1024*8*4*2,  
        "buffer_size": 100000,
        "learning_rate": 0.0002,
        "action_noise": "normal",
        "gradient_steps": 500,
        "train_freq": (5000, "step"), 
        "learning_starts": 10,
        "ent_coef": "auto_0.1",
    }

    POLICY_KWARGS = dict(net_arch=dict(pi=[256, 128, 128, 64, 64], qf=[256, 128, 128, 64, 64]))

model_sac_before_train = agent.get_model("sac", seed=46, model_kwargs=SAC_PARAMS, policy_kwargs=POLICY_KWARGS)

the result is as below
Episode 6 end successful last = 1107
day_start = 5
---------------------------------
| time/              |          |
|    episodes        | 20       |
|    fps             | 335      |
|    time_elapsed    | 64       |
|    total_timesteps | 21540    |
| train/             |          |
|    actor_loss      | 3.67e+14 |
|    critic_loss     | 1.27e+29 |
|    ent_coef        | 1.15e+13 |
|    ent_coef_loss   | -848     |
|    learning_rate   | 0.0002   |
|    n_updates       | 150500   |
|    reward          | 119      |
---------------------------------
Episode 7 end successful last = 1025

day_start = 97
---------------------------------
| time/              |          |
|    episodes        | 248      |
|    fps             | 317      |
|    time_elapsed    | 827      |
|    total_timesteps | 262556   |
| train/             |          |
|    actor_loss      | 1.2e+15  |
|    critic_loss     | 1.41e+30 |
|    ent_coef        | 3.84e+13 |
|    ent_coef_loss   | -882     |
|    learning_rate   | 0.0002   |
|    n_updates       | 156500   |
|    reward          | 7.04     |
---------------------------------
Episode 64 end successful last = 1104

it look like everything grow bigger, why?
if the stock data is pure random, can i get zero loss in train?i think i can, but i will get a bad result in test part.

**Comment by Athe-kunal** [comment]
You have a batch size of 65536, which is very big. What is the reasoning behind it? Also, what does auto_0.1 for entropy coefficient means? I am not aware of it.

**Comment by war3gu** [comment]
> You have a batch size of 65536, which is very big. What is the reasoning behind it? Also, what does auto_0.1 for entropy coefficient means? I am not aware of it.

i change the super params as below
    env_kwargs_train = {
        "stock_dim": stock_dimension,
        "hmax": 1000,
        "initial_amount": 1000000,
        "buy_cost_pct": 6.87e-5,
        "sell_cost_pct": 1.0687e-3,
        "reward_scaling": 1e0,
        "state_space": state_space,
        "action_space": stock_dimension,
        "tech_indicator_list": config.TECHNICAL_INDICATORS_LIST,
        "print_verbosity": 100,                 
        "initial_buy": True,
        "hundred_each_trade": True,
        "out_of_cash_penalty": 0.001,
        "cash_limit": 0.2,
        "model_name":"stock_a",
        "mode":"train",                 
        "random_start":True
    }

  
    SAC_PARAMS = {
        "batch_size": 1024,   
        "buffer_size": 100000,
        "learning_rate": 0.0002,
        "action_noise": "normal",
        "gradient_steps": 100,    
        "train_freq": (1000, "step"),  
        "learning_starts": 10,
        "ent_coef": "auto_0.1",
    }

    POLICY_KWARGS = dict(net_arch=dict(pi=[256, 128, 128, 64, 64], qf=[256, 128, 128, 64, 64]))

the result is as below
---------------------------------
| time/              |          |
|    episodes        | 4        |
|    fps             | 759      |
|    time_elapsed    | 5        |
|    total_timesteps | 4264     |
| train/             |          |
|    actor_loss      | 769      |
|    critic_loss     | 2.35e+05 |
|    ent_coef        | 0.101    |
|    ent_coef_loss   | 64.6     |
|    learning_rate   | 0.0002   |
|    n_updates       | 100      |
|    reward          | 0        |
---------------------------------

---------------------------------
| time/              |          |
|    episodes        | 16       |
|    fps             | 767      |
|    time_elapsed    | 22       |
|    total_timesteps | 17284    |
| train/             |          |
|    actor_loss      | 768      |
|    critic_loss     | 1.47e+05 |
|    ent_coef        | 0.107    |
|    ent_coef_loss   | 62.9     |
|    learning_rate   | 0.0002   |
|    n_updates       | 400      |
|    reward          | 23.1     |
---------------------------------
---------------------------------
| time/              |          |
|    episodes        | 472      |
|    fps             | 750      |
|    time_elapsed    | 665      |
|    total_timesteps | 499640   |
| train/             |          |
|    actor_loss      | 2.18e+03 |
|    critic_loss     | 1.11e+05 |
|    ent_coef        | 1.18     |
|    ent_coef_loss   | -4.72    |
|    learning_rate   | 0.0002   |
|    n_updates       | 12400    |
|    reward          | 17.5     |
---------------------------------
this is the first loop result
the last loop result is as below
---------------------------------
| time/              |          |
|    episodes        | 400      |
|    fps             | 748      |
|    time_elapsed    | 565      |
|    total_timesteps | 423104   |
| train/             |          |
|    actor_loss      | 2.75e+12 |
|    critic_loss     | 9.97e+22 |
|    ent_coef        | 4.87e+09 |
|    ent_coef_loss   | -629     |
|    learning_rate   | 0.0002   |
|    n_updates       | 123000   |
|    reward          | 0        |
---------------------------------

everything grow to a super big value
auto_0.1 means that  ent_coef start from 0.1
you can get my code from   https://github.com/war3gu/FinRL-Meta
run the file Demo_China_A_share_market_sac.py
my python version is 3.7

**Comment by Athe-kunal** [comment]
Your model has diverged I guess. So make sure your dataset does not have any NaN values and you can try to tune the hyperparameters rather than fixating them. You can use some of the tutorials in the repo for tuning
